### PR TITLE
chore: update nydusd snapshotter dep

### LINKED ISSUE #7: Support the registry with token based authorization
An error is thrown when run with `podman run -it --tls-verify=false --rm ghcr.io/dragonflyoss/image-service/ubuntu:nydus-nightly-v5 /bin/bash`:

```
ERRO[2022-08-16T03:45:58.792583759Z] download snapshot files failed: failed to GET request with code 401
failed to resolve from ref ghcr.io/dragonflyoss/image-service/ubuntu:nydus-nightly-v5, digest sha256:12eb2b017edffb9c3cdc4e7f1387535cd9fb634321fe08ea157a074b65c39067
github.com/containerd/nydus-snapshotter/pkg/filesystem/fs.(*Filesystem).PrepareMetaLayer
	/home/imeoer/go/pkg/mod/github.com/containerd/nydus-snapshotter@v0.3.0-alpha.4/pkg/filesystem/fs/fs.go:491
github.com/containers/nydus-storage-plugin/pkg/manager.(*LayerManager).ResolverMetaLayer
	/home/imeoer/nydus-storage-plugin/pkg/manager/manager.go:151
github.com/containers/nydus-storage-plugin/pkg/fs.(*layerNode).Lookup
	/home/imeoer/nydus-storage-plugin/pkg/fs/layer_node.go:86
github.com/hanwen/go-fuse/v2/fs.(*rawBridge).lookup
	/home/imeoer/go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.1.0/fs/bridge.go:340
github.com/hanwen/go-fuse/v2/fs.(*rawBridge).Lookup
	/home/imeoer/go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.1.0/fs/bridge.go:323
github.com/hanwen/go-fuse/v2/fuse.doLookup
	/home/imeoer/go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.1.0/fuse/opcode.go:333
github.com/hanwen/go-fuse/v2/fuse.(*Server).handleRequest
	/home/imeoer/go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.1.0/fuse/server.go:483
github.com/hanwen/go-fuse/v2/fuse.(*Server).loop
	/home/imeoer/go/pkg/mod/github.com/hanwen/go-fuse/v2@v2.1.0/fuse/server.go:456
runtime.goexit
	/usr/lib/go-1.18/src/runtime/asm_amd64.s:1571
```

### PR DESCRIPTION
**Pull request by fatelei**
- fix #7  auth support anonymous auth
- update github action using prebuild nydus format image instead of run registry locally

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -274,6 +272,8 @@ github.com/envoyproxy/go-control-plane v0.9.9-0.20210512163311-63b5d3c536b0/go.m
 github.com/envoyproxy/go-control-plane v0.9.10-0.20210907150352-cf90f659a021/go.mod h1:AFq3mo9L8Lqqiid3OhADV3RfLJnjiw63cSpi+fDTRC0=
 github.com/envoyproxy/protoc-gen-validate v0.1.0/go.mod h1:iSmxcyjqTsJpI2R4NaDN7+kN2VEUnK/pcBlmesArF7c=
 github.com/evanphx/json-patch v4.9.0+incompatible/go.mod h1:50XU6AFN0ol/bzJsmQLiYLvXMP4fmwYFNcr97nuDLSk=
+github.com/fatelei/nydus-snapshotter v0.3.0-alpha.7 h1:3FVE44etcBcDnPcDlaBLDSGZEhmU4XJKQO+pQs6xObw=
```
Comment (imeoer) [inline go.sum:275]: Maybe we can submit a PR to `containerd/nydus-snapshotter` first. :)

Comment (imeoer) [reply go.sum:275]: Someday we should remove the `containerd/nydus-snapshotter` dep.

Comment (fatelei) [reply go.sum:275]: `containerd/nydus-snapshotter` has remove `PrepareMetaLayer` method, so we can using `main` branch

**Review by imeoer — APPROVED** [review]
LGTM

**Pull request merged**